train_segmentation.py

```diff
@@ -519,7 +514,10 @@ def train(cfg: DictConfig) -> None:
 
     # MODEL PARAMETERS
     class_weights = get_key_def('class_weights', cfg['dataset'], default=None)
-    loss_fn = get_key_def('loss_fn', cfg['training'], default='CrossEntropy')
+    loss_fn = cfg.loss
+    if not loss_fn.is_binary == (num_classes == 1):
+        raise ValueError(f"A binary loss was chosen for a multiclass task")
+    del loss_fn.is_binary  # prevent exception at instantiation
```

Comment:
@victorlazio109 this is the simplest way I've found to make sure binary losses are not called in multiclass problem and vice versa.